docs/reference/search/search-your-data/semantic-search.asciidoc

```diff
@@ -0,0 +1,131 @@
+[[semantic-search]]
+== Semantic search
+
+Semantic search is a search method that helps you find data based on the intent
+and contextual meaning of a search query, instead of a match on query terms
+(lexical search).
+
+{es} provides semantic search capabilities using {ml-docs}/ml-nlp.html[natural
+language processing (NLP)] and vector search. Deploying an NLP model to {es}
+enables it to extract text embeddings out of text. Embeddings are vectors that
+provide a numeric representation of a text. Pieces of content with similar
+meaning have similar representations. 
```

Comment:
I am wondering whether it would be a good idea to say even at this early stage that there are two different ways of performing semantic search:

1. Using dense vector embeddings and vector search
2. Using sparse vector embeddings and a weighted terms search

Maybe it's too early to mention this in the introductory paragraph, but it's very important within Elasticsearch because the way you use the two types of models are very different.

I've asked for other opinions on Slack, so maybe someone will think of a compelling reason to go one way or the other.